bsp4j/src/main/java/ch/epfl/scala/bsp4j/CppExtension.xtend

```diff
@@ -0,0 +1,43 @@
+package ch.epfl.scala.bsp4j
+
+import java.util.List
+import org.eclipse.lsp4j.jsonrpc.validation.NonNull
+import org.eclipse.lsp4j.generator.JsonRpcData
+
+@JsonRpcData
+class CppBuildTarget {
+  @NonNull CppPlatform platform
+  String cCompiler
+  String cppCompiler
+  new(@NonNull CppPlatform platform, String cCompiler, String cppCompiler) {
+    this.platform = platform
+    this.cCompiler = cCompiler
+    this.cppCompiler = cppCompiler
+  }
+}
+
+@JsonRpcData
+class CppDependenciesSourcesParams {
+  @NonNull List<BuildTargetIdentifier> targets
+  String originId
+  new(@NonNull List<BuildTargetIdentifier> targets) {
+    this.targets = targets
+  }
+}
+
+@JsonRpcData
+class CppDependenciesSourcesResult {
+  @NonNull List<CppDependenciesSourcesItem> items
+  new(@NonNull List<CppDependenciesSourcesItem> items) {
+    this.items = items
+  }
+}
+
+@JsonRpcData
+class CppDependenciesSourcesItem {
+  @NonNull BuildTargetIdentifier target
+  String packageNamePrefix
```

Comment:
Given that this request for now exists just for getting this prefix, should it be in the constructor? I'm  not very keen on doing that if there is ever a need for more data on it.